src/Tools/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/AnalyzerTest`1.cs

```diff
@@ -98,10 +98,10 @@ public string TestCode
         public Dictionary<string, string> XmlReferences { get; } = new Dictionary<string, string>();
 
         /// <summary>
-        /// Gets or sets a value indicating whether exclusions for generated code should be tested automatically. The
-        /// default value is <see langword="true"/>.
+        /// Gets or sets a value the analysis exclusions to verify. The default value is a combination of
+        /// <see cref="AnalysisExclusions.GeneratedCode"/> and <see cref="AnalysisExclusions.Suppression"/>.
         /// </summary>
-        public bool VerifyExclusions { get; set; } = true;
+        public AnalysisExclusions Exclusions { get; set; } = AnalysisExclusions.GeneratedCode | AnalysisExclusions.Suppression;
```

Comment:
I find `AnalysisExclusions.Suppression` a peculiar flag to include as a default, since I've never written or seen an analyzer that proactively looks for `#pragma` and turns itself off. But if Roslyn is doing the filtering for the analyzer, why is this one even an option, since an analyzer can't avoid it?

Reply:
I guess I still find this API confusing then. Does `Exclusions` means that the verifier will not test it, or that it *should* test that the analyzer *omits* diagnostics for the identified scenario?
Is there another word that doesn't leave the ambiguity? Something like `VerifyNoDiagnosticsFor` ?